File: python-app/state.py
"""Tiny shared state between telegram_worker.py and server.py, avoiding a
circular import between the two."""

_monitored_sources = []


def set_monitored_sources(sources):
    global _monitored_sources
    _monitored_sources = sources


def get_monitored_sources():
    return _monitored_sources

File: server/src/state.js
// Tiny shared state so index.js can publish the resolved monitored-channel
// list for the /api/sources route to read.

let monitoredSources = [];

function setMonitoredSources(sources) {
  monitoredSources = sources;
}

function getMonitoredSources() {
  return monitoredSources;
}

module.exports = { setMonitoredSources, getMonitoredSources };
